collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormsTask.java

```diff
@@ -33,32 +38,51 @@
  * @author carlhartung
  */
 public class DownloadFormsTask extends
-        AsyncTask<ArrayList<ServerFormDetails>, String, HashMap<ServerFormDetails, String>> implements FormDownloaderListener {
+        AsyncTask<ArrayList<ServerFormDetails>, String, Map<ServerFormDetails, String>> {
```

Comment:
Ah it's weirdness with `AsyncTask` where `execute` takes a varargs of `Params`. Generally to avoid how consfusing this ends up looking I'd leave `Params` as `Void` and pass anything I need into the constructor (as you can't reuse an `AsyncTask` anyway). I think we can leave this for the moment as it's existing and ideally we want to just get rid of the `AsyncTask` rather than improve it.